spec/models/reservation_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe Reservation, type: :model do
+  before(:all) do
+    @user = User.create(username: "john doe1#{SecureRandom.hex}", email: "jod#{SecureRandom.hex}@test.com", password: '123456')
+    expect(@user).to be_valid
+    @doctor = Doctor.create!(name: 'doc1', detail: 'doctor details1', photo: 'some url1', city: 'some city1', specialization: 'specialization1', fee: 100.00)
+    @reservation = Reservation.new(city: 'Imphal', date: '23/8/2022', user_id: @user.id, doctor_id: @doctor.id)
+  end
```

Comment:
Great job with this preparation of users, doctor, and reservation as setup to perform your tests! 💯